File: swift_5.x/sample/Sources/sample/sample.swift
// Example 1: Addition
typealias Add = (Int) -> (Int) -> Int
let add: Add = { (a: Int) in { (b: Int) in a + b } }
let addThree = add(3)
print(addThree(4))

// Example 2: String Prefixing
typealias Prefix = (String) -> (String) -> String
let addPrefix = { (a: String) in { (b: String) in "[\(a)] \(b)" } }
let prefixWithWarning = addPrefix("WARNING")
print(prefixWithWarning("All is well."))

/**
OUTPUT:
$ swift Sources/sample/sample.swift
> 7
> [WARNING] All is well.
*/

File: elixir/sample/lib/sample.ex
defmodule Sample do
  @moduledoc """
  Documentation for Sample.
  """

  @doc """
  Hello world.

  ## Examples

      iex> Sample.addThree(4)
      7

      iex> Sample.prefixWithWarning("All is well.")
      "[WARNING] All is well."

  """
  # Example 1: Addition
  def add a do
    fn b -> a + b end
  end

  def addThree b do
    Sample.add(3).(b)
  end

  # Example 2: String Prefixing
  def addPrefix a do
    fn b -> EEx.eval_string("[<%= a %>] <%= b %>", a: a, b: b) end
  end

  def prefixWithWarning a do
    Sample.addPrefix("WARNING")
  end
end
